src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -2383,3 +2438,18 @@ fn strip_generics_from_path_segment(
         Err(ResolutionFailure::MalformedGenerics(MalformedGenerics::UnbalancedAngleBrackets))
     }
 }
+use std::{num::ParseFloatError, str::FromStr};
+
+/// Uses the rules from [`f64::from_str()`].
+pub struct Number {
+    pub value: f64,
+}
+
+impl FromStr for Number {
+    type Err = ParseFloatError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let value = s.parse::<f64>()?;
+        Ok(Self { value })
+    }
+}
```

Comment:
I just left here so the example from the issue can be tested by building the docs for this module:

```
./x.py doc --open src/librustdoc
```

Will remove once we are finished with the feedback/review.
```suggestion
```

Reply:
Please remove it now and make it a test in `src/test/rustdoc/intra-doc`.

You can see how it looks with `./x.py build rustdoc` and then [`rustdoc +stage2 path/to/test.rs`](https://rustc-dev-guide.rust-lang.org/rustdoc.html#cheat-sheet).